lambda/kendra-webcrawler-status/kendra_webcrawler_status.py

```diff
@@ -48,31 +48,41 @@ def get_data_source_id(index_id, data_source_name):
 
 
 def kendra_list_data_source_sync_jobs(IndexId, data_source_id):
-    response = client.list_data_source_sync_jobs(
+    #get information about a Kendra index data source
+    response = client.describe_data_source(
         Id=data_source_id,
-        IndexId=IndexId,
+        IndexId=IndexId
     )
+    data_source_status = response["Status"]
 
+    #get information on the data sync jobs for a given data source
+    response = client.list_data_source_sync_jobs(
+        Id=data_source_id,
+        IndexId=IndexId
+    )
     # get current status by sorting the result by start time descending order
-    # if status is equal to status in var status list, the status will be pending
-    pending_statuses = [
-        "SYNCING",
-        "STOPPING",
-        "SYNCING_INDEXING",
-    ]
-
-    latest_history_item = functools.reduce(lambda x, y: x if x['StartTime'] > y['StartTime'] else y, response['History'])
-    status = "PENDING" if latest_history_item['Status'] in pending_statuses else "SUCCEEDED"
-    print(status)
-    result = list(map(lambda item: {'StartTime': item['StartTime'].strftime("%m/%d/%Y, %H:%M:%S"),
+    if response['History'] != []:   #if there is data sync history
+        latest_history_item = functools.reduce(lambda x, y: x if x['StartTime'] > y['StartTime'] else y, response['History'])
+
+        result = list(map(lambda item: {'StartTime': item['StartTime'].strftime("%m/%d/%Y, %H:%M:%S"),
                                     'EndTime': item['EndTime'].strftime("%m/%d/%Y, %H:%M:%S") if 'EndTime' in item else '',
                                     'Status': item['Status'] if item['Status'] != "INCOMPLETE" else "COMPLETE WITH ERRORS",
                                     'ErrorMessage': item['ErrorMessage'] if 'ErrorMessage' in item else '',
                                     'Metrics': item['Metrics']
                                     }, response['History']))
+        status = latest_history_item['Status']
```

Comment:
Are we overloading the `status` of the data source,

-  `'Status': 'CREATING'|'DELETING'|'FAILED'|'UPDATING'|'ACTIVE',`

 and the `status` of data source sync job (from the latest history of the sync job)

-  `'Status': 'FAILED'|'SUCCEEDED'|'SYNCING'|'INCOMPLETE'|'STOPPING'|'ABORTED'|'SYNCING_INDEXING',`

And the superset of values is returned in the response of this function?

Reply:
We are using two different status(es) for this -- Data Source status, and Data Source Sync Job status.
But yes, there is room here to make this more clearer in the UI, when displaying status(es). Lets review this more.